app/CMakeLists.txt

```diff
@@ -52,119 +39,94 @@ option(SANITIZE_THREAD    "Sanitize Thread"        OFF)
+target_include_directories(TetWild PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/tetwild)
+# 	target_link_libraries(TetWild PUBLIC FastEnvelope)
+target_compile_features(TetWild PUBLIC ${CXX14_FEATURES})
+
+if(TWTK_APPS_WITH_SANITIZERS)
+	add_sanitizers(TetWild)
+endif()
+
+add_executable(TetWild_bin tetwild/main.cpp)
+target_link_libraries(TetWild_bin PUBLIC TetWild)
+
+
+
+################################################################################
+# Quadric Edge Collapse 2D
+################################################################################
+
+add_library(QuadricEdgeCollapse
+		external/Exception.h
+		external/MshSaver.cpp
+		external/MshSaver.h
 
-		EdgeSplitting.cpp
-		EdgeCollapsing.cpp
+		Logger.cpp
+		Logger.hpp
 		)
 
+target_link_libraries(QuadricEdgeCollapse PUBLIC igl::predicates WildMeshingToolkit warnings::all)
+target_include_directories(QuadricEdgeCollapse PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+
+target_compile_features(QuadricEdgeCollapse PUBLIC ${CXX14_FEATURES})
+
+if(TWTK_APPS_WITH_SANITIZERS)
+	add_sanitizers(QuadricEdgeCollapse)
+endif()
+
+add_executable(QuadricEdgeCollapse_bin QuadricEdgeCollapse/main.cpp)
+target_link_libraries(QuadricEdgeCollapse_bin PUBLIC QuadricEdgeCollapse)
+
+
 ################################################################################
-# Required libraries
+# Quadric Edge Collapse 2D
 ################################################################################
 
-# if(FLOAT_TETWILD_USE_FLOAT)
-# 	target_compile_definitions(${PROJECT_NAME} PUBLIC -DFLOAT_TETWILD_USE_FLOAT)
-# endif()
-
-# target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC ${GMP_INCLUDE_DIRS})
-
-target_link_libraries(${PROJECT_NAME} PUBLIC igl::predicates WildMeshingToolkit
-		# ${GMP_LIBRARIES}
-		# ${MPFR_LIBRARIES}
-)
-# if(FLOAT_TETWILD_ENABLE_TBB)
-# 	target_link_libraries(${PROJECT_NAME} PUBLIC tbb::tbb)
-# 	target_compile_definitions(${PROJECT_NAME} PUBLIC FLOAT_TETWILD_USE_TBB)
-# endif()
-
-# if(FLOAT_TETWILD_WITH_EXACT_ENVELOPE)
-# 	target_link_libraries(${PROJECT_NAME} PUBLIC FastEnvelope)
-# 	target_compile_definitions(${PROJECT_NAME} PUBLIC NEW_ENVELOPE)
-# endif()
-
-####
-
-if(FLOAT_TETWILD_TOPLEVEL_PROJECT)
-	add_executable(${PROJECT_NAME}_bin main.cpp)
-	target_compile_features(${PROJECT_NAME}_bin PUBLIC ${CXX14_FEATURES})
-
-	target_link_libraries(${PROJECT_NAME}_bin
-			PUBLIC
-			${PROJECT_NAME}
-			warnings::all
-			)
-
-	if(TARGET igl::tetgen)
-		target_link_libraries(${PROJECT_NAME} PUBLIC igl::tetgen)
-		target_compile_definitions(${PROJECT_NAME} PUBLIC -DLIBIGL_WITH_TETGEN)
-	endif()
-
-	if(FLOAT_TETWILD_WITH_SANITIZERS)
-		add_sanitizers(${PROJECT_NAME}_bin)
-	endif()
-
-	if(NOT (${CMAKE_VERSION} VERSION_LESS "3.6.0"))
-		set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT ${PROJECT_NAME}_bin)
-	endif()
-endif()
\ No newline at end of file
+include(CTest)
+enable_testing()
```

Comment:
Since CMake 3.15, `include(CTest)` will call `enable_testing()`, so only 1 line is needed (but we'd need to update our min CMake version).

Also, `include(CTest)` should be set once at the root CMakeLists.txt.

Reply:
done!, changed the required cmake to 3.15